File: output/deepseek_r1/avatar/algo_based_translation/python/java/atcoder_AGC019_A.java

import java.util.Scanner;

public class atcoder_AGC019_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int q = scanner.nextInt();
        int h = scanner.nextInt();
        int s = scanner.nextInt();
        int d = scanner.nextInt();
        int n = scanner.nextInt();
        
        int best1L = Math.min(Math.min(4 * q, 2 * h), s);
        int best2L = Math.min(d, 2 * best1L);
        
        int pairs = n / 2;
        int totalCost = n % 2 == 0 
            ? pairs * best2L 
            : pairs * best2L + best1L;
        
        System.out.println(totalCost);
    }
}

File: output/llama_4_maverick/avatar/algo_based_translation/java/python/failed_translation/atcoder_AGC019_A.py

Q, H, S, D, N = map(int, input().split())
onePrice = min(Q * 4, H * 2, S)
if onePrice <= D / 2:
    total_cost = N * onePrice
else:
    total_cost = (N // 2) * D + (N % 2) * onePrice
print(total_cost)

